### PR TITLE
Add customer extensions mount points and pass object IDs to extensions

### PR DESCRIPTION
**Pull request by krzysztofwolski**
I want to merge this change because there is a new app mount that has to be handled.

Additional change: app popups displayed on the detail pages will receive object id. For example: the product details extension popup will get the id of the currently viewed product, so it will be able to display data related to it. Since it's an additional query param, it is a nonbreaking change for the apps.  

How to test:

Added example app which use all available mounting points: https://github.com/saleor/saleor-app-template/pull/106

1. start dashboard dev server with this branch, connect it to saleor cloud instance
2. clone the example app
3. `pnpm i`, `pnpm dev` the app to start it
4. in the app directory run `saleor app tunnel`
5. now you can go to the dev dashboard and witness full glory of available app extensions

### Pull Request Checklist

1. [ ] This code contains UI changes
6. [ ] All visible strings are translated with proper context including data-formatting
7. [ ] Attributes `[data-test-id]` are added for new elements
8. [x] Changes are mentioned in the changelog
9. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants

CONTAINERS=1

### CODE REVIEW

**Review by dominik-zeglen — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

File: src/utils/urls.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import isArray from "lodash/isArray";
 import { stringify } from "qs";
 
-export function stringifyQs(params: {}): string {
+export function stringifyQs(params: {}, arrayFormat?: string): string {
   return stringify(params, {
-    arrayFormat: "indices",
+    arrayFormat: arrayFormat || "indices",
```
Comment (krzysztofwolski) [inline src/utils/urls.ts:6]: Optional argument added, to change a way how array URLs are passed to apps:
was: list[0]=aaa$list[1]=bbb
now: list=aaa,bbb

File: src/apps/components/AppFrame/AppFrame.tsx [hunk 2]
```diff
@@ -100,7 +106,10 @@ export const AppFrame: React.FC<Props> = ({
       src={urlJoin(
         src,
         window.location.search,
-        `?domain=${shop.domain.host}&id=${appId}&locale=${locale}`,
+        `?${stringifyQs(
+          { domain: shop.domain.host, id: appId, ...params },
+          "comma",
+        )}`,
```
Comment (andrzejewsky) [inline src/apps/components/AppFrame/AppFrame.tsx:112]: Could you move that to a separate function? let's keep building URL logic in one place

**Review by andrzejewsky — APPROVED** [review]

File: src/apps/components/AppFrame/AppFrame.tsx (hunk 2 above)
Comment (krzysztofwolski) [reply src/apps/components/AppFrame/AppFrame.tsx]: Updated. WDYT?

Comment (andrzejewsky) [reply src/apps/components/AppFrame/AppFrame.tsx]: how about moving this to `@saleor/apps/urls` ?

Comment (krzysztofwolski) [reply src/apps/components/AppFrame/AppFrame.tsx]: @andrzejewsky updated

File: src/customers/components/CustomerListPage/CustomerListPage.tsx [hunk 3]
```diff
@@ -48,27 +67,50 @@ const CustomerListPage: React.FC<CustomerListPageProps> = ({
...
-        <Button
-          variant="primary"
-          href={customerAddUrl}
+      <PageHeader
+        title={intl.formatMessage(sectionNames.customers)}
+        cardMenu={
+          extensionMenuItems.length > 0 && (
+            <CardMenu
+              className={classes.settings}
+              menuItems={[...extensionMenuItems]}
```
Comment (Droniu) [inline src/customers/components/CustomerListPage/CustomerListPage.tsx:98]: Why is this dereferenced?

**Review by Droniu — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

File: src/customers/components/CustomerListPage/CustomerListPage.tsx (hunk 3 above)
Comment (krzysztofwolski) [reply src/customers/components/CustomerListPage/CustomerListPage.tsx]: Resolved

**Review by Droniu — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

**Pull request merged**